Command

vor-o-na opened the pull request.

Commit:
Poprawienie nazwy klasy

Commit:
Próba zmiany operateTo(naprawianie wzoru)

vor-o-na committed:
Wyświetlenie symulacji w oknie aplikacji

vor-o-na committed:
Refactor & second test figure

vor-o-na committed:
DriverAdapter->DrawLineAdapter

vor-o-na committed:
3.3

vor-o-na committed:
3.2.3 change

vor-o-na committed:
zaimplemantowana hierarchia+execute()+ComplexCommand

vor-o-na committed:
ComplexCommand edit

vor-o-na committed:
create draw funcs

vor-o-na committed:
create command classes; edit execute();

vor-o-na committed:
finish Command

kowallus commented on src/main/java/edu/kis/powp/jobs2d/command/CommandFactory.java:
```diff
@@ -0,0 +1,7 @@
+import java.util.List;
+
+public abstract class CommandFactory {
+    public abstract ComplexCommand draw();
```
Draw is a wrong name. The command instance is returned (not drawn)

kowallus commented on src/main/java/edu/kis/powp/jobs2d/command/ComplexCommand.java:
```diff
@@ -0,0 +1,27 @@
+    public void add(DriverCommand command) {
+        commands.add(command);
+    }
+    public void remove(DriverCommand command) {
+        commands.remove(command);
+    }
+    public List<DriverCommand> getCommands() {
+        return commands;
+    }
```
not necessary

kowallus commented on src/main/java/edu/kis/powp/jobs2d/command/SmallSquareFactory.java:
```diff
@@ -0,0 +1,14 @@
+
+public class SmallSquareFactory extends CommandFactory{
+    @Override
+    public ComplexCommand draw() {
```
DriverCommand return type would be sufficient.

kowallus commented on src/test/java/edu/kis/powp/jobs2d/events/SelectTestFigureOptionListener.java:
```diff
@@ -20,10 +25,17 @@ public SelectTestFigureOptionListener(DriverManager driverManager) {
+		}else if (command.contains(FIGURE3)){
+			square.draw().execute(driverManager.getCurrentDriver());
+		}else if (command.contains(FIGURE4)){
+			smallSquare.draw().execute(driverManager.getCurrentDriver());
```
driverManager.getCurrentDriver duplicated several times

kowallus commented on src/test/java/edu/kis/powp/jobs2d/events/SelectTestFigureOptionListener.java:
```diff
@@ -20,10 +25,17 @@ public SelectTestFigureOptionListener(DriverManager driverManager) {
 	@Override
 	public void actionPerformed(ActionEvent e) {
 		String command=e.getActionCommand();
```
Menu action should not depend on its caption.

kowallus commented on src/test/java/edu/kis/powp/jobs2d/events/SelectTestFigureOptionListener.java:
```diff
@@ -11,6 +14,8 @@
 	private final String FIGURE1 = "1";
 	private final String FIGURE2 = "2";
+	private final String FIGURE3 = "3";
+	private final String FIGURE4 = "4";
```
Public constants would make more sense

kowallus reviewed (changes requested):
1.5 pts (too late penalty)

too pass please fix most of the issues listed in this and adapter PR.

vor-o-na committed:
Command fixed:
-CommandFactory: func name changed
-ComplexCommand: removed not necessary funcs
-SelectTestFigureOptionListener: duplications fixed
-SelectTestFigureOptionListener: private->public

vor-o-na committed:
Adapter fixed:
-MyLineDrawerAdapter->SpecialLineDrawerDriverAdapter
-DottedLineDrawerDriverAdapter: added Dottedline mode

vor-o-na commented:
Command and Adapter fixed

kowallus commented on src/main/java/edu/kis/powp/jobs2d/drivers/adapter/DottedLineDrawerDriverAdapter.java:
```diff
@@ -0,0 +1,37 @@
+import edu.kis.legacy.drawer.shape.LineFactory;
+import edu.kis.powp.jobs2d.Job2dDriver;
+
+public class DottedLineDrawerDriverAdapter extends DrawPanelController implements Job2dDriver{
```
Inheritance is not used (no super drawLine calls). Should be removed!

kowallus commented on src/main/java/edu/kis/powp/jobs2d/drivers/adapter/SpecialLineDrawerDriverAdapter.java:
```diff
@@ -5,11 +5,11 @@
 import edu.kis.powp.jobs2d.Job2dDriver;
 
-public class MyLineDrawerAdapter extends DrawPanelController implements Job2dDriver {
+public class SpecialLineDrawerDriverAdapter extends DrawPanelController implements Job2dDriver {
```
Inheritance is not used (no super drawLine calls). Should be removed!

kowallus commented on src/main/java/edu/kis/powp/jobs2d/drivers/adapter/DottedLineDrawerDriverAdapter.java:
```diff
@@ -0,0 +1,37 @@
+
+    @Override
+    public void operateTo(int x, int y) {
+        ILine line = LineFactory.getDottedLine();
```
Hardcoded line type. What about 3.3.2 test?
ILine instance should be configured using Dependency Injection (i.e., field and constructor/setter)

kowallus commented on src/test/java/edu/kis/powp/jobs2d/events/SelectTestFigureOptionListener.java:
```diff
@@ -3,19 +3,18 @@
-	private final String FIGURE2 = "2";
-	private final String FIGURE3 = "3";
-	private final String FIGURE4 = "4";
+	public final String FIGURE1 = "1";
+	public final String FIGURE2 = "2";
+	public final String FIGURE3 = "3";
+	public final String FIGURE4 = "4";
```
Constants should be used in TestJobs2dPattern to define the names of tests. (This is later used below, String command=e.getActionCommand();)

kowallus reviewed (changes requested):
Several small issues still to fix.

vor-o-na committed:
3.3.2 test + minor changes

kowallus commented on src/test/java/edu/kis/powp/jobs2d/events/SelectTestFigureOptionListener.java:
```diff
@@ -27,15 +25,19 @@ public void actionPerformed(ActionEvent e) {
-			smallSquare.createCommand().execute(currDriver);
-
+		switch (command) {
+			case "Figure 1":
+				FiguresJoe.figureScript1(currDriver);
+				break;
+			case "Figure 2":
+				FiguresJoe.figureScript2(currDriver);
+				break;
+			case "Figure 3":
+				square.createCommand().execute(currDriver);
+				break;
+			case "Figure 4":
```
case labels should refer to the same constants introduced in TestJobs2Patterns

kowallus reviewed (commented):
3- pts for individual part
